src/components/modals/ConditionModal.tsx

```diff
@@ -134,13 +138,13 @@ const Component: React.FC<Props> = (props) => {
     const [showNumberValue, setShowNumberValue] = React.useState(true)
     const [numberValue, setNumberValue] = React.useState(0)
     const [enumValueOptions, setEnumValueOptions] = React.useState<EnumOption[]>([])
+    const [selectedEnumValueOption, setSelectedEnumValueOption] = React.useState<EnumOption>()
```

Comment:
Again, what rule caused this. Hooks are meant to be co-located so they can be moved as custom hook together in future.  Eg. all the enum functionality is grouped. Would change this back.

Reply:
This is the "use before definition" lint also.